tests/test_cross_validation.py

```diff
@@ -53,5 +56,40 @@ def test_cv_pancancer(expression_data, cancer_type):
             sample_info_df, test_df_pancancer.index)
 
     assert set([cancer_type]).issubset(train_cancer_types)
-    assert len(train_cancer_types) > len(set([cancer_type])) 
+    assert len(train_cancer_types) > len(set([cancer_type]))
     assert test_cancer_types == set([cancer_type])
+
+
+def test_stratified_cv(expression_data):
+    rnaseq_df, sample_info_df = expression_data
+    sample_info_df = sample_info_df.reindex(rnaseq_df.index)
+    num_folds = 4
+    train_proportions = []
+    test_proportions = []
+
+    for fold in range(num_folds):
+        train_df, test_df, stratify_df = du.split_stratified(
+            rnaseq_df, sample_info_df, num_folds=4, fold_no=fold
+        )
+
+        assert train_df.shape[0] + test_df.shape[0] == rnaseq_df.shape[0]
+
+        train_df = train_df.merge(stratify_df, left_index=True, right_index=True)
+        test_df = test_df.merge(stratify_df, left_index=True, right_index=True)
+        train_counts = train_df.id_for_stratification.value_counts()
+        test_counts = test_df.id_for_stratification.value_counts()
+        train_fold_props = train_counts / train_counts.sum()
+        test_fold_props = test_counts / test_counts.sum()
+        train_proportions.append(train_fold_props)
+        test_proportions.append(test_fold_props)
+
+    # check that proportions of each stratification are approximately
+    # the same between folds (in other words, check that stratified CV is
+    # working properly)
+    #
+    # note that the absolute scale of the proportions can vary quite a bit,
+    # but on a relative scale they should be pretty close
+    for ix1, ix2 in it.combinations(range(num_folds), 2):
+        assert np.allclose(train_proportions[ix1], train_proportions[ix2], rtol=1.0)
+        assert np.allclose(test_proportions[ix1], test_proportions[ix2], rtol=1.0)
+        assert np.allclose(train_proportions[ix1], test_proportions[ix2], rtol=1.0)
```

Comment:
Since `np.allclose()` isn't symmetric, it may make sense to use `it.permutations` here, but the edge case where the test passes because the second index's values are large seems unlikely if you have more than ~2 cancer types

Reply:
Good call, I'll change it